Resolve comparison of different signedness warnning

```
/__w/OpenMLDB/OpenMLDB/src/apiserver/api_server_impl.cc:657:28: warning: comparison of integer expressions of different signedness: ‘int’ and ‘size_t’ {aka ‘long unsigned int’} [-Wsign-compare]
         for (auto i = 0; i < size; i++) {
                          ~~^~~~~~
/__w/OpenMLDB/OpenMLDB/src/apiserver/api_server_impl.cc:694:18: warning: comparison of integer expressions of different signedness: ‘size_t’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
         if (size != schema.size()) return ar;
```

refer https://github.com/4paradigm/OpenMLDB/actions/runs/3134931174/jobs/5090039479#step:7:358

--- a/src/apiserver/api_server_impl.cc
+++ b/src/apiserver/api_server_impl.cc
@@ -689,9 +689,9 @@ JsonReader& operator&(JsonReader& ar, std::shared_ptr<openmldb::sdk::SQLRequestR
     int32_t str_length = 0;
     {
         ar.Member("data");
-        int size;
+        size_t size;
         ar.StartArray(&size);  // start first iter "data"
-        if (size != schema.size()) return ar;
+        if (static_cast<int>(size) != schema.size()) return ar;
 
         for (auto col = schema.begin(); col != schema.end(); col++) {
             if (col->type() == ::hybridse::type::kVarchar) {

--- a/src/apiserver/api_server_impl.cc
+++ b/src/apiserver/api_server_impl.cc
@@ -654,7 +654,7 @@ JsonReader& operator&(JsonReader& ar, std::shared_ptr<openmldb::sdk::SQLRequestR
         ar.Member("schema");
         size_t size;
         ar.StartArray(&size);  // start "schema"
-        for (auto i = 0; i < size; i++) {
+        for (size_t i = 0; i < size; i++) {
             std::string type;
             ar& type;
             // uppercase
@@ -689,7 +689,7 @@ JsonReader& operator&(JsonReader& ar, std::shared_ptr<openmldb::sdk::SQLRequestR
     int32_t str_length = 0;
     {
         ar.Member("data");
-        size_t size;
+        int size;
         ar.StartArray(&size);  // start first iter "data"
         if (size != schema.size()) return ar;
 

